\documentclass{article}
\usepackage[utf8]{inputenc}
\usepackage[colorinlistoftodos]{todonotes}
\usepackage{setspace}
\usepackage[margin = 1in]{geometry}
\usepackage{multirow}
\usepackage[framemethod=TikZ]{mdframed}
\usepackage{newfloat}
\usepackage{multicol}

\usepackage{amsmath}
\usepackage[
    backend=biber,
    style=authoryear,
    natbib=true,
    url=true, 
    doi=true,
    eprint=true
]{biblatex}
\addbibresource{research.bib}
\usepackage{graphicx}
\usepackage{subcaption}
\usepackage{array}
\usepackage{amssymb}
\usepackage{amsthm}
\usepackage{amsfonts}
\usepackage{longtable}
\usepackage[outdir=./]{epstopdf}
\DeclareCaptionType{equ}[][] 
%https://stackoverflow.com/questions/149479/adding-a-caption-to-an-equation-in-latex
%\captionsetup[equ]{labelformat=empty}

\usepackage{hyperref}
\hypersetup{
    colorlinks=true
}
\DeclareFloatingEnvironment[fileext=frm,placement={!ht},name=Frame]{myfloat}
\captionsetup[myfloat]{labelfont=bf}
\newenvironment{frameenv}[1]
    {\begin{myfloat}[tb]
    \begin{mdframed}[roundcorner=10pt,backgroundcolor=blue!10]
    \caption{#1}
    %\begin{multicols*}{2}
    }
    {%\end{multicols*}
    \end{mdframed}\end{myfloat}
    }




\title{Testing Pleiotropy vs. Separate QTL in Multiparental Populations}
\author{Frederick Boehm}
\date{\today}



\begin{document}
\doublespacing
\maketitle
\tableofcontents


%%%%% 3A start
\section{Expression hotspot dissection}
\subsection{Introduction}

A central goal of systems genetics studies is to identify causal relationships between genetic variants and quantitative traits. Recent work by \citet{chick2016defining} has popularized linear regression-based methods for causal inference in systems genetics. The great successes of these methods in identifying causal relationships among genetic variants and quantitative biomolecule concentrations motivates our goal of clarifying a role for our test of pleiotropy vs. separate QTL in systems genetics studies. We argue that our test of pleiotropy vs. separate QTL complements mediation analysis in two ways. First, our test limits the set of candidate mediators by ruling out traits that don't share a single pleiotropic QTL. Second, when regression-based mediation analysis fails to identify a mediator, our test of pleiotropy vs. separate QTL still provides information on the number of QTL. The number of separate QTL may provide clues to aid biological understanding.

We begin below with an overview of a causal mediation analysis framework that \citet{vanderweele2015explanation} summarizes. We then discuss applications of parts of this framework to the systems genetics context. We describe in our methods section the study design and statistical analyses in which we use mediation analysis and our pleiotropy vs. separate QTL test to dissect an expression trait hotspot with data from 378 Diversity Outbred mice.






Specifically, one may formulate a mediation analysis by identifying a marker (and its founder allele probabilities) as the "driver" of a causal diagram. In recalling the central dogma of molecular biology, we want to consider a marker that affects transcript levels of a nearby gene \citep{crick1958protein}. For example, if a gene is located on Chromosome 2 and centered at 100 Mb, then we would want to consider markers that are also on Chromosome 2 and near 100 Mb, perhaps between 98 Mb and 102 Mb. If a second expression trait - typically transcripts from a gene that is not “local” - maps near the marker that we’ve identified, then we can ask whether trait1 “mediates” the relationship between trait2 and the marker.

To determine whether trait1 mediates the relationship between trait2 and the marker genotypes, \emph{i.e.}, the driver, we perform a series of regression analyses, which we detail below (Equations \ref{model1}, \ref{model2}, \ref{model3}, \ref{model4}). In brief, we perform regression-based QTL mapping between trait2 and the marker, with and without conditioning on the candidate mediator. If the LOD score diminishes sufficiently upon conditioning on a candidate mediator, then we declare the candidate mediator a mediator.

The rationale behind this strategy is as follows. If the driver affects trait2 solely by way of trait1, then conditioning on trait1 would nullify the relationship between driver and trait2. At the other extreme, if the driver affects trait2 solely through mechanisms that don't involve trait1, then conditioning on trait1 values would not affect the association between the driver and trait2.

One role for our test of pleiotropy vs separate QTL is to potentially limit the set of candidate mediators that require study. For example, if trait1 and trait2 map to separate QTL, then it would be unlikely that trait1 affects trait2 (or vice versa). On the other hand, if we believe that a single pleiotropic locus affects both trait1 and trait2, then a mediation analysis is particularly useful in efforts to clarify the relationship between trait1 and trait2. In this setting, where a single pleiotropic locus affects both trait1 and trait2, and we seek to identify whether the trait1 or trait2 is the intermediate, we would perform two mediation analyses, one with trait1 as candidate intermediate between driver and trait2, and a second analysis with trait2 as a candidate intermediate between driver and trait1.

%\subsection{Regression-based mediation methods}

Frequently one uses regression-based methods to identify causal intermediates from among multiple measured variables \citep{baron1986moderator}. For example, \citet{chick2016defining} measured 16,921 gene transcript levels and 6,756 protein concentrations in liver tissue from 192 mice. After identifying QTL for gene transcript levels and protein concentrations, they performed regression-based mediation analyses. For each candidate mediator, they fitted four linear models:

\begin{equ}[!ht]
\begin{equation}
Y = b1 + WC + E
\label{model1}
\end{equation}
\caption{Linear model with intercept and covariates only.}
\end{equ}

\begin{equ}[!ht]
\begin{equation}
Y = XB + WC + E
\label{model2}
\end{equation}
\caption{Linear model with founder allele dosages and covariates.}
\end{equ}

\begin{equ}[!ht]
\begin{equation}
Y = b1 + WC + M\beta + E
\label{model3}
\end{equation}
\caption{Linear model with intercept, covariates, and candidate mediator.}
\end{equ}

\begin{equ}[!ht]
\begin{equation}
Y = XB + WC + M\beta + E
\label{model4}
\end{equation}
\caption{Linear model with founder allele dosages, covariates, and candidate mediator.}
\end{equ}


In the above four models, $X$ is a $n$ by $8$ matrix of founder allele dosages at a single marker, $B$ is a $8$ by $1$ matrix of founder allele effects, $E$ is a $n$ by $1$ matrix of random errors, $b$ is an number, $1$ is a $n$ by $1$ matrix with all entries set to $1$, $Y$ is a $n$ by $1$ matrix of phenotype values (for a single trait), and $M$ is a $n$ by $1$ matrix of values for a putative mediator. $C$ is a matrix of covariate effects, and $W$ is a matrix of covariates. We denote the coefficient of the mediator by $\beta$.

We assume that the vector $E$ is (multivariate) normally distributed with zero vector as mean and covariance matrix $\Sigma = \sigma^2I_n$, where $I_n$ is the $n$ by $n$ identity matrix.

With the above models, the log-likelihoods are easily calculated. For example, in Equation \ref{model1}, the vector $Y$ follows a multivariate normal distribution with mean $(b1 + WC)$ and covariance $\Sigma = \sigma^2I$. Thus, we can write the likelihood for Model \ref{model1} as:

\begin{equation}
    L(b, C, \sigma^2| Y, W) = (2\pi)^{- \frac{n}{2}}\exp{ \left(- \frac{1}{2}(Y - b1 - WC)^T\Sigma^{-1}(Y - b1 - WC)\right)}
\end{equation}

We thus have the following equation for the log-likelihood for model 1:

\begin{equation}
    \log L(b, C, \sigma^2 | Y, W) = - \frac{n}{2}\log (2\pi) - \frac{1}{2} (Y - b1 - WC)^T\Sigma^{-1}(Y - b1 - WC)
\end{equation}


\citet{chick2016defining} calculated the log$_{10}$ likelihoods for all four models before determining two LOD scores:
\begin{equation}
LOD_1 = log_{10}(\text{model 2 likelihood}) - log_{10}(\text{model 1 likelihood})
\end{equation}

\begin{equation}
LOD_2 = log_{10}(\text{model 4 likelihood}) - log_{10}(\text{model 3 likelihood})
\end{equation}

And, finally, \citet{chick2016defining} calculated the statistic:

\begin{equation}
\text{LOD difference} = LOD_1 - LOD_2
\end{equation}

LOD difference need not be positive. For example, in the setting where the putative mediator is not a true mediator, $LOD_1$ and $LOD_2$ values may lead to a negative LOD difference statistic. In our analyses in this section, we truncated these negative LOD difference values to zero.

For each mediation analysis, we consider the LOD difference proportion, which we define below (Equation \ref{eq:lod-diff-prop}).

\begin{equation}
\text{LOD difference proportion} = \frac{(LOD_1 - LOD_2)}{LOD_1}
\label{eq:lod-diff-prop}
\end{equation}

In other words, we consider what proportion of the association strength, on the LOD scale, is diminished by conditioning on a putative mediator. This statistic differs from the LOD difference statistic in that it scales the mediation LOD difference by LOD$_1$ value. We do this in efforts to accommodate the diversity of LOD peak heights in our data. While a value of 5 for LOD difference may be important for a trait with a LOD$_1$ of 12, another trait with a LOD$_1$ of 120 that has a LOD$_2$ of 115, suggests that nearly all of the signal remains transmitted when conditioning on the putative mediator.

To perform a mediation analysis, we must specify three components of a possible directed acyclic graph. For our purposes, we always begin with a marker's founder allele dosages. Since we are working in the context of dissecting an expression QTL hotspot, we choose the ``driver'' to be the founder allele dosages at the peak position for a local expression trait. We then choose the terminus of the putative graph, i.e., the ``target'', to be a nonlocal expression trait.






%\subsection{Four assumptions for causal inference}

To claim that the LOD difference statistic reflects a causal relationship, four assumptions about confounding are needed \citep{vanderweele2015explanation}.


%\begin{mdframed}
%\caption{XXXX}
%\begin{enumerate}
%\item No unmeasured confounding of the treatment-outcome %relationship
%\item No unmeasured confounding of the mediator-outcome %relationship
%\item No unmeasured confounding of the treatment-mediator relationship
%\item No mediator-outcome confounder that is affected by the treatment
%\label{list4}
%\end{enumerate}
%\end{mdframed}

\begin{frameenv}{Four assumptions for causal inference}\label{frame1}
  \begin{enumerate}
\item No unmeasured confounding of the treatment-outcome relationship
\item No unmeasured confounding of the mediator-outcome relationship
\item No unmeasured confounding of the treatment-mediator relationship
\item No mediator-outcome confounder that is affected by the treatment
\end{enumerate}

\end{frameenv}






In the above terminology, ``treatment'' refers to the founder allele dosages, while ``outcome'' is the nonlocal gene expression trait, and ``mediator'' is the local gene expression trait.





%\subsection{Declaring mediators}

Multiple research teams have proposed strategies for assessing statistical significance of the "mediation LOD difference" statistic. \citet{chick2016defining} used individual transcript levels as sham mediators and tabulated their LOD difference statistics. They then compared the observed LOD difference statistics for putative mediators to the empirical distribution of LOD difference statistics obtained from the collection of sham mediators. \citet{keller2018genetic}, on the other hand, in their study of pancreatic islet cell biology, declared mediators those local transcripts that diminished the LOD score of nonlocal transcripts by at least 1.5 \citep{keller2018genetic}. While significance threshold determination remains an active area of research, we proceed below by examining all 147 nonlocal transcript levels that \citet{keller2018genetic} identified as mapping to the Chromosome 2 hotspot.



\subsection{Methods}

We examined the potential that the two methods - 1. pleiotropy vs. separate QTL testing and 2. mediation analysis - could play complementary roles in efforts to dissect gene expression trait hotspots. After describing our data, we explain our statistical analyses involving 13 local gene expression traits and 147 nonlocal gene expression traits, all of which map to a 4-Mb hotspot on Chromosome 2.

%\subsection{Data description}

We analyzed data from 378 Diversity Outbred mice \citep{keller2018genetic}. \citet{keller2018genetic} genotyped tail biopsies with the GigaMUGA microarray \citep{morgan2015mouse}. They also used RNA sequencing to measure genome-wide pancreatic islet cell gene expression for each mouse at the time of sacrifice \citep{keller2018genetic}. They shared these data, together with inferred founder allele probabilities, on the Data Dryad site (\url{https://datadryad.org/resource/doi:10.5061/dryad.pj105}).

We examined the Chromosome 2 pancreatic islet cell expression trait hotspot that \citet{keller2018genetic} identified. \citet{keller2018genetic} identified 147 nonlocal traits that map to the 4-Mb region centered at 165.5 Mb on Chromosome 2. The 147 nonlocal traits all exceeded the genome-wide LOD significance threshold, 7.18 \citep{keller2018genetic}. With regression-based mediation analyses, they identified expression levels of local gene \emph{Hnf4a} as a mediator of 88 of these 147 nonlocal traits.

Because \citet{keller2018genetic} reported that some nonlocal traits that map to the Chromosome 2 hotspot did not demonstrate evidence of mediation by \emph{Hnf4a} expression levels, we elected to study a collection of local gene expression traits, rather than \emph{Hnf4a} alone. This strategy enabled us to ask whether one of twelve other local traits mediates those nonlocal hotspot traits that are not mediated by \emph{Hnf4a}. Our set of local gene expression traits includes \emph{Hnf4a} and 12 other local genes (Table \ref{tab:annot}). Our 13 local genes are the only genes that met all of the following criteria:

\begin{enumerate}
    \item QTL peak with LOD $>$ 40
    \item QTL peak position within 2 Mb of hotspot center (165.5 Mb)
    \item Gene midpoint within 2 Mb of hotspot center (165.5 Mb)
\end{enumerate}

The 147 nonlocal traits that we studied all had LOD peak heights above 7.18 and LOD peak positions within 2 Mb of the center of the hotspot (at 165.5 Mb).



%\subsection{Statistical analyses}

We used both univariate and bivariate QTL scans. In the univariate scans, we identified LOD peaks for each of 160 (147 nonlocal and 13 local) expression traits that mapped to within 2 Mb of the center of the Chromosome 2 expression trait hotspot. 



%\subsubsection{Univariate QTL scans}

\citet{keller2018genetic} provided univariate QTL mapping results in their Data Dryad file, so we didn't repeat the univariate QTL scans. We briefly summarize their analyses below.

For a given univariate phenotype, \citet{keller2018genetic} fitted a linear mixed effects model (with polygenic random effect) at every marker across the genome. \citet{keller2018genetic} performed calculations with the \texttt{qtl2} R package \citep{Broman2018}.



%\subsubsection{Bivariate QTL scans}

Our analyses involved 13 local expression traits and 147 nonlocal expression traits. We described above the criteria for choosing these expression traits.

We performed a series of two-dimensional QTL scans in which we paired each local gene's transcript levels with each nonlocal gene's transcript levels, for a total of 13 * 147 = 1911 two-dimensional scans. Each scan examined the same set of 180 markers, which spanned the interval from 163.1 Mb to 167.8 Mb and included univariate peaks for all 13 + 147 = 160 expression traits. We performed these analyses with the R package \texttt{qtl2pleio} \citep{qtl2pleio}.

For each QTL scan, we fitted a collection of bivariate models for all 180 * 180 = 32,400 ordered pairs of markers. Throughout our analyses, we treated the founder allele probabilities as known, despite the fact that they are inferred through hidden Markov model methods \citep{broman2012genotype, broman2012haplotype, Broman2018}.

For each ordered pair of markers, we fitted a multivariate linear mixed effects model with the methods of Chapter 2.

%\subsubsection{Mediation analyses}

We performed linear regression-based mediation analyses for all 1,911 local-nonlocal trait pairs. We probed the extent to which each nonlocal trait's association strength diminished upon conditioning on transcript levels of a putative mediator. Each of the 13 local expression traits, considered one at a time, served as putative mediators. We thus fitted the four linear regression models that we describe above (Equations \ref{model1}, \ref{model2}, \ref{model3}, \ref{model4}).

One question that needs clarification is the choice of ``driver'' for each mediation analysis. We elected to use the founder allele probabilities at the marker that demonstrated the univariate LOD peak for the putative mediator. Alternative analyses, in which one chooses as driver the founder allele probabilities at which the nonlocal trait has its univariate peak, are also possible.



%\subsubsection{Local gene analyses}

To visualize the summary statistics for our two methods, we plotted, for each local gene, a scatterplot of LOD difference proportion values against pleiotropy vs. separate QTL test statistics.

% latex table generated in R 3.5.1 by xtable 1.8-3 package
% Mon Dec 17 10:04:32 2018
\begin{table}[ht]
\centering
\begin{tabular}{lrrrr}
  \hline
symbol & start & end & LOD\_peak\_position & LOD\_peak\_height \\
  \hline
Pkig & 163.66 & 163.73 & 163.52 & 51.68 \\
  Serinc3 & 163.62 & 163.65 & 163.58 & 126.93 \\
  Hnf4a & 163.51 & 163.57 & 164.02 & 48.98 \\
  Stk4 & 164.07 & 164.16 & 164.03 & 60.39 \\
  Pabpc1l & 164.03 & 164.05 & 164.03 & 52.50 \\
  Slpi & 164.35 & 164.39 & 164.61 & 40.50 \\
  Neurl2 & 164.83 & 164.83 & 164.64 & 64.58 \\
  Cdh22 & 165.11 & 165.23 & 165.05 & 53.84 \\
  2810408M09Rik & 165.49 & 165.49 & 165.57 & 67.34 \\
  Eya2 & 165.60 & 165.77 & 165.72 & 98.89 \\
  Prex1 & 166.57 & 166.71 & 166.75 & 46.91 \\
  Ptgis & 167.19 & 167.24 & 167.27 & 56.25 \\
  Gm14291 & 167.20 & 167.20 & 167.27 & 73.72 \\
   \hline
\end{tabular}
\caption{Local gene annotations for analysis of Chromosome 2 expression trait hotspot. All positions are in units of Mb on Chromosome 2. LOD peak position and LOD peak height refer to those obtained from univariate analyses. ``Start'' and ``end'' refer to the local gene's DNA start and end positions, as annotated by Ensembl version 75.}
\label{tab:annot}
\end{table}

%\subsubsection{Nonlocal gene analyses}

We also examined the 1911 pairs from the per-nonlocal gene perspective. For each of the 147 nonlocal genes, we plotted LOD difference proportion against pleiotropy test statistic values. We present below (Figure \ref{fig:4nonlocal}) examples that illustrate some of the observed patterns between pleiotropy test statistics and LOD difference proportion values.


\subsection{Results}

%\subsection{Scatter plot for all 1911 pairs}

We present below our scatter plot for all 147 * 13 = 1911 pairs of traits (Figure \ref{fig:lod-diff-prop-v-lrt-all}). Each pair contains one local expression trait and one nonlocal expression trait. Each point in the figure represents a single pair. We see that points with high values of LOD difference proportion tend to have small values of pleiotropy test statistic, and those points with high values of the pleiotropy test statistic tend to have small values of LOD difference proportion.

We colored blue the points that involve \emph{Hnf4a} as the local gene; all other local genes have red points. The most striking feature of the coloring is that many blue points have small values of the pleiotropy test statistics and very high values of LOD difference proportion.



\begin{figure}
    \centering
    \includegraphics[width = 0.7\textwidth]{../keller2018-chr2-hotspot-chtc/Rmd/lod-diff-prop-v-lrt.eps}
    \caption[LOD difference proportion vs. pleiotropy test statistic]{LOD difference proportion against pleiotropy vs. separate QTL test statistics for 1911 pairs of traits. Each point represents a single pair. Pairs that involve local expression trait Hnf4a are colored blue, while all others are colored red. Note the high prevalence of blue points in the upper left quadrant of the figure. These points, with low values of the pleiotropy vs. separate QTL test statistic and high values of LOD difference proportion, are consistent with \emph{Hnf4a} transcript levels mediating the effect of \emph{Hnf4a} genetic variants affecting nonlocal transcript abundances.}
    \label{fig:lod-diff-prop-v-lrt-all}
\end{figure}

%\subsection{Local gene analyses}

To more thoroughly examine the relationships across the 13 local genes, we created 13 plots of LOD difference proportion (from the mediation analyses) against pleiotropy test statistic. They reveal common patterns. First, we see no points in the upper right quadrant of each plot. This tells us that those nonlocal genes with high values of pleiotropy test statistic have low values of LOD difference proportion. Similarly, those nonlocal genes with high values of LOD difference proportion tend to have small values of the pleiotropy test statistic. Finally, some trait pairs demonstrate low values of both the LOD difference proportion and pleiotropy test statistic. This observation suggests that, for a given local expression trait, these pairs are not mediated by the local expression trait yet they arise from a shared pleiotropic locus.

In comparing the \emph{Hnf4a} plot (Figure \ref{fig:hnf4a}) with the other 12 plots (Figure \ref{fig:nothnf4a-12}), we see that none of the 12 plots in Figure \ref{fig:nothnf4a-12} closely resembles Figure \ref{fig:hnf4a}. \emph{Serinc3}, \emph{Stk4}, \emph{Neurl2}, and \emph{Cdh22} are closest in appearance to the plot of \emph{Hnf4a}. However, each of \emph{Serinc3}, \emph{Stk4}, \emph{Neurl2}, and \emph{Cdh22} has very few points with LOD difference proportion above 0.5, while \emph{Hnf4a} has many points with LOD difference proportion above 0.5.

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{../keller2018-chr2-hotspot-chtc/Rmd/Hnf4a-lod-diff-prop-v-lrt.eps}
    \caption[LOD difference proportion vs. pleiotropy test statistic.]{Scatter plot of LOD difference proportion against pleiotropy vs. separate QTL test statistic for 147 pairs of traits. Each pair includes \emph{Hnf4a} and one of the nonlocal gene expression traits that map to the Chromosome 2 hotspot.}
    \label{fig:hnf4a}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width = 0.8\textwidth]{../keller2018-chr2-hotspot-chtc/Rmd/12local-facet_grid-no-strip.eps}
    \caption[LOD difference proportion vs. pleiotropy test statistic from the per-local gene perspective.]{Scatter plots of LOD difference proportion against pleiotropy vs. separate QTL test statistic with 147 pairs of traits per panel. Each panel includes a local gene expression trait (per the label in the upper right quadrant) and one of the 147 nonlocal gene expression traits that map to the Chromosome 2 hotspot.}
    \label{fig:nothnf4a-12}
\end{figure}






% latex table generated in R 3.5.1 by xtable 1.8-3 package
% Wed Dec 19 10:50:32 2018
\begin{table}[ht]
\centering
\begin{tabular}{lc}
  \hline
 Local gene & Number of nonlocal gene expression traits \\
  \hline
2810408M09Rik &   3 \\
  Cdh22 &   4 \\
  Eya2 &   4 \\
  Gm14291 &   5 \\
  Hnf4a &  89 \\
  Neurl2 &  15 \\
  Pabpc1l &   4 \\
  Pkig &   2 \\
  Prex1 &   5 \\
  Ptgis &   3 \\
  Serinc3 &   7 \\
  Slpi &   1 \\
  Stk4 &   5 \\
   \hline
\end{tabular}
\caption{Number of nonlocal gene expression traits for which each local gene expression trait is the strongest mediator.}
\label{tab:med-count}
\end{table}

%\subsection{Nonlocal gene analyses}

Scatter plots of LOD difference proportion values against pleiotropy vs. separate QTL test statistics for each of the 147 nonlocal expression traits demonstrated multiple patterns. Eighty-nine nonlocal genes' plots showed \emph{Hnf4a} to have the greatest value, among the 13 putative mediators, of the LOD difference proportion statistic. In many cases, \emph{Hnf4a}'s LOD difference proportion statistic was at least twice that of any of the other 12 local gene expression levels.

\begin{figure}
    \centering
    \caption{Scatter plots for four nonlocal expression traits. Each plot features 13 points, one for each local gene expression trait. The y axis denotes LOD difference proportion values, while the x axis corresponds to pleiotropy vs. separate QTL test statistics. Blue points represent the pairing with local gene expression trait \emph{Hnf4a}. Red points represent the other 12 local gene expression traits.}
    \begin{subfigure}[t]{.45\textwidth}
        \includegraphics[width = \textwidth]{../keller2018-chr2-hotspot-chtc/Rmd/nonlocal-scatter_97.eps}
        \caption[\emph{Hnf4a} transcript levels mediate \emph{Abcb4} transcript levels.]{\emph{Hnf4a} transcript levels mediate \emph{Abcb4} transcript levels. The high LOD difference proportion and the very small pleiotropy test statistic together provide evidence that \emph{Hnf4a} mediates \emph{Abcb4} transcript levels.}
    \end{subfigure}
    \begin{subfigure}[t]{.45\textwidth}
        \includegraphics[width = \textwidth]{../keller2018-chr2-hotspot-chtc/Rmd/nonlocal-scatter_105.eps}
        \caption[\emph{Hnf4a} transcript levels mediate \emph{Smim5} transcript levels.]{\emph{Hnf4a} transcript levels mediate \emph{Smim5} transcript levels. The high LOD difference proportion and the very small pleiotropy test statistic together provide evidence that \emph{Hnf4a} mediates \emph{Smim5} transcript levels.}
    \end{subfigure}
    \begin{subfigure}[t]{.45\textwidth}
        \includegraphics[width = \textwidth]{../keller2018-chr2-hotspot-chtc/Rmd/nonlocal-scatter_64.eps}
        \caption[\emph{Tmprww4} transcript levels are not mediated by any of the 13 local gene expression traits.]{\emph{Tmprww4} transcript levels are not mediated by any of the 13 local gene expression traits. However, several local gene expression traits arise from separate QTL, as evidenced by their large (greater than 5) values of the pleiotropy test statistic.}
    \end{subfigure}
    \begin{subfigure}[t]{.45\textwidth}
        \includegraphics[width = \textwidth]{../keller2018-chr2-hotspot-chtc/Rmd/nonlocal-scatter_141.eps}
        \caption[\emph{Gm3095} transcript levels are not mediated by any of the 13 local genes.]{\emph{Gm3095} transcript levels are not mediated by any of the 13 local genes. The tests of pleiotropy demonstrate evidence of separate QTL for some local expression traits when paired with \emph{Gm3095}.}
    \end{subfigure}
    \label{fig:4nonlocal}
\end{figure}





\subsection{Discussion}

Our pairwise analyses with both mediation analyses and tests of pleiotropy vs. separate QTL provide additional evidence for the importance of \emph{Hnf4a} in the biology of the Chromosome 2 hotspot in pancreatic islet cells. Our analyses, and, specifically, the test of pleiotropy vs. separate QTL, may be more useful when studying nonlocal traits that map to a hotspot yet don't show strong evidence of mediation by local expression traits. In such a setting, the test of pleiotropy vs. separate QTL can, at least, provide some information about the genetic architecture at the hotspot. Specifically, our test of pleiotropy vs. separate QTL may inform inferences about the number of underlying QTL in a given expression trait hotspot. Additionally, our test may limit the number of expression traits that are potential intermediates between a QTL and a specified nonlocal expression trait. This relies on the assumption that an intermediate expression trait and a target expression trait presumably share a QTL.

On the other hand, mediation analyses, when they provide evidence for mediation of a nonlocal trait by a local expression trait, are more informative than the test of pleiotropy vs. separate QTL, since the mediation analyses identify precisely the intermediate expression trait.

We recommend using both tests of pleiotropy vs separate QTL and mediation analyses when dissecting an expression trait QTL hotspot. In practice, mediation analyses, which involve a collection of univariate regressions at a small set of pre-specified markers, are less computationally intense than tests of pleiotropy vs. separate QTL, since the latter requires a two-dimensional QTL scan over the region of interest. This two-dimensional QTL scan often involves fitting more than ten thousand bivariate regression models. In light of the computational cost of testing pleiotropy vs separate QTL, future researchers may wish to use it as a follow-up to mediation analyses when examining expression trait hotspots.

Future research may investigate the use of polygenic random effects in the statistical models for mediation analysis. Additional methodological questions include approaches for declaring significant a mediation LOD difference proportion and consideration of other possible measures and scales of extent of mediation. Additionally, future researchers may wish to consider biological models that contain two mediators.

The social and health sciences have witnessed much methods research in  mediation analysis. The field of statistical genetics has not fully adopted these strategies yet, but, given the nature of current and future data, many opportunities exist for translation of approaches from epidemiology to systems genetics. For example, the 2015 text from Vander Weele contains detailed discussions of many methods issues that arose in mediation analyses in epidemiology studies.





% latex table generated in R 3.5.1 by xtable 1.8-3 package
% Tue Dec 18 18:18:00 2018
{\tiny
\begin{longtable}{lrr}
\caption{LOD peak positions and peak heights for 147 expression traits that map to the Chromosome 2 expression trait hotspot.}\\
\hline \\
%\begingroup\tiny

Gene symbol & Peak position & LOD \\
  \hline
Mtfp1 & 163.52 & 17.79 \\
  Slc12a7 & 163.58 & 8.30 \\
  Gpa33 & 163.58 & 26.38 \\
  Tmem25 & 163.58 & 11.26 \\
  Klhl29 & 163.58 & 11.03 \\
  Slc9a3r1 & 163.58 & 8.40 \\
  Kif1c & 163.58 & 11.12 \\
  Gata4 & 163.58 & 8.72 \\
  Ppp2r5b & 163.58 & 7.80 \\
  Vil1 & 163.58 & 26.48 \\
  Aldh4a1 & 163.58 & 9.38 \\
  Cotl1 & 163.58 & 16.25 \\
  Tmprss4 & 163.58 & 18.30 \\
  Fhod3 & 163.58 & 17.62 \\
  Zfp750 & 163.58 & 7.83 \\
  Svop & 163.58 & 10.12 \\
  Abcb4 & 163.58 & 16.59 \\
  Ccnjl & 163.58 & 6.70 \\
  Sephs2 & 163.58 & 27.21 \\
  Pcdh1 & 163.58 & 12.48 \\
  Fat1 & 163.58 & 7.49 \\
  Sox4 & 163.58 & 14.71 \\
  Gm8206 & 163.58 & 11.33 \\
  Gm8492 & 163.58 & 10.47 \\
  Clcn5 & 164.02 & 10.49 \\
  Slc6a8 & 164.02 & 8.35 \\
  Bcmo1 & 164.02 & 47.31 \\
  Arrdc4 & 164.02 & 8.75 \\
  Cacnb3 & 164.02 & 47.06 \\
  Sec14l2 & 164.02 & 12.00 \\
  Pgrmc1 & 164.02 & 15.88 \\
  Baiap2l2 & 164.02 & 20.33 \\
  Recql5 & 164.02 & 33.97 \\
  Cpd & 164.02 & 13.70 \\
  Degs2 & 164.02 & 15.20 \\
  Muc13 & 164.02 & 18.91 \\
  Clic5 & 164.02 & 10.39 \\
  Tff3 & 164.02 & 16.38 \\
  Myo7b & 164.02 & 13.70 \\
  Afg3l2 & 164.02 & 19.12 \\
  Sema4g & 164.02 & 23.39 \\
  Agap2 & 164.02 & 33.99 \\
  Plxna2 & 164.02 & 8.61 \\
  Aldob & 164.02 & 20.99 \\
  Epb4.1l4b & 164.02 & 9.21 \\
  Sel1l3 & 164.02 & 17.92 \\
  Sult1b1 & 164.02 & 17.63 \\
  Hpgds & 164.02 & 31.14 \\
  Ush1c & 164.02 & 21.90 \\
  Calml4 & 164.02 & 12.62 \\
  Fam83b & 164.02 & 16.12 \\
  Myo15b & 164.02 & 71.57 \\
  Inpp5j & 164.02 & 11.57 \\
  Ttyh2 & 164.02 & 15.20 \\
  Cdhr2 & 164.02 & 30.00 \\
  Myrf & 164.02 & 37.55 \\
  Sh3bp4 & 164.02 & 9.04 \\
  Vgf & 164.02 & 15.05 \\
  Grtp1 & 164.02 & 23.88 \\
  B4galnt3 & 164.02 & 20.43 \\
  Gucy2c & 164.02 & 12.02 \\
  Smim5 & 164.02 & 18.20 \\
  Nrip1 & 164.02 & 9.50 \\
  Clrn3 & 164.02 & 20.13 \\
  Acot4 & 164.02 & 12.17 \\
  Hunk & 164.02 & 18.50 \\
  Zbtb16 & 164.02 & 6.69 \\
  Osgin1 & 164.02 & 13.51 \\
  Zfp541 & 164.02 & 25.28 \\
  2610042L04Rik & 164.02 & 8.14 \\
  Gm9429 & 164.02 & 9.13 \\
  Agxt2 & 164.02 & 14.54 \\
  Gm17147 & 164.02 & 26.94 \\
  Gm8281 & 164.02 & 8.40 \\
  Ddx23 & 164.03 & 9.29 \\
  Map2k6 & 164.03 & 18.54 \\
  Npr1 & 164.03 & 9.68 \\
  Hdac6 & 164.03 & 11.28 \\
  Vav3 & 164.03 & 13.51 \\
  Atp11b & 164.03 & 11.77 \\
  Aadat & 164.03 & 8.08 \\
  Tmem19 & 164.03 & 20.60 \\
  Gbp4 & 164.03 & 7.14 \\
  Papola & 164.03 & 12.14 \\
  Als2 & 164.03 & 18.35 \\
  Sult1d1 & 164.03 & 11.23 \\
  Myo6 & 164.03 & 10.27 \\
  Dnajc22 & 164.03 & 13.72 \\
  Unc5d & 164.03 & 6.96 \\
  Gm3095 & 164.03 & 10.90 \\
  Gm26886 & 164.03 & 8.76 \\
  Eps8 & 164.06 & 12.49 \\
  Ddc & 164.06 & 13.61 \\
  Fras1 & 164.06 & 10.06 \\
  Card11 & 164.06 & 7.20 \\
  Glyat & 164.06 & 10.79 \\
  Pipox & 164.08 & 15.90 \\
  Iyd & 164.08 & 23.23 \\
  Man1a & 164.26 & 8.77 \\
  Cdc42ep4 & 164.26 & 8.11 \\
  Ppara & 164.26 & 8.84 \\
  Galr1 & 164.26 & 10.10 \\
  Ctdsp1 & 164.26 & 7.13 \\
  Eif2ak3 & 164.26 & 8.93 \\
  Misp & 164.26 & 9.24 \\
  Sun1 & 164.26 & 7.35 \\
  Kctd8 & 164.26 & 9.76 \\
  Dcaf12l1 & 164.26 & 8.95 \\
  4930539E08Rik & 164.26 & 8.25 \\
  Slc29a4 & 164.26 & 7.94 \\
  Gm3239 & 164.26 & 9.21 \\
  Gm3629 & 164.26 & 9.77 \\
  Gm3252 & 164.26 & 9.87 \\
  Gm3002 & 164.26 & 7.62 \\
  Ctsh & 164.29 & 9.76 \\
  Dao & 164.29 & 7.06 \\
  Ak7 & 164.31 & 10.07 \\
  Pcp4l1 & 164.35 & 9.11 \\
  Gm12929 & 164.62 & 37.80 \\
  Mgat1 & 164.63 & 10.10 \\
  Atg7 & 164.76 & 7.82 \\
  Gm11549 & 165.05 & 10.23 \\
  Ccdc111 & 165.12 & 7.77 \\
  Fam20a & 165.15 & 7.59 \\
  Oscp1 & 165.16 & 7.99 \\
  Dsc2 & 165.28 & 8.06 \\
  Adam10 & 165.28 & 9.73 \\
  Plb1 & 165.28 & 8.85 \\
  Ccdc89 & 165.28 & 7.84 \\
  9930013L23Rik & 165.28 & 7.93 \\
  Gm13648 & 165.28 & 7.08 \\
  Igfbp4 & 165.45 & 8.68 \\
  Ldlrap1 & 165.45 & 7.18 \\
  Cdh18 & 165.45 & 7.96 \\
  Arhgef10l & 165.46 & 7.57 \\
  Cib3 & 165.48 & 8.70 \\
  Macf1 & 165.57 & 9.10 \\
  Fam63b & 165.58 & 8.33 \\
  1190002N15Rik & 165.63 & 9.02 \\
  Bcl2l14 & 165.71 & 7.54 \\
  Hist2h2be & 165.88 & 8.96 \\
  Gm6428 & 165.89 & 7.32 \\
  Dennd5b & 166.18 & 9.42 \\
  Gm12168 & 166.18 & 7.77 \\
  Gm12230 & 166.42 & 7.46 \\
  Acat3 & 166.61 & 7.17 \\
  Gpr20 & 166.84 & 8.41 \\
   \hline
  % \endgroup
  
  \label{tab:hot-annot}

\end{longtable}
}
%%%%% 3A end

%%%% 3B start
\section{Power analyses}

\subsection{Introduction}

The goal of this section is to characterize the statistical power of our pleiotropy test under a variety of conditions by studying a real data set. We examine pancreatic islet expression traits from the \citet{keller2018genetic} data. As in chapters 2 and 3A, we test only two traits at a time. Because we’ve chosen local expression traits in our analysis, we both know where each trait’s true QTL location (approximately), and we anticipate that each trait has a unique QTL that is distinct from QTL for other local expression traits. This design thus provides opportunities to study statistical power for our test.

We anticipate that inter-locus distance, univariate QTL strength, and correlation of founder allele effects patterns are three factors that contribute to power for our test. Specifically, we expect that greater inter-locus distance, greater univariate LOD scores, and less similar founder allele effects patterns correspond to greater statistical power to detect two separate QTL.

We use pancreatic islet gene expression traits from a publicly available data set, which \citet{keller2018genetic} first collected, analyzed, and shared. We examine a collection of 80 local traits on Chromosome 19 and perform our test for pleiotropy on pairs of traits. We also examine pairwise relationships among gene expression traits to characterize the impacts of univariate LOD score, inter-locus distance, and similarity of founder allele effects patterns on pleiotropy test statistics.



\subsection{Methods}


We analyzed data from 378 Diversity Outbred mice \citep{keller2018genetic}. \citet{keller2018genetic} genotyped tail biopsies with the GigaMUGA microarray \citep{morgan2016mouse}. They used RNA sequencing to measure genome-wide pancreatic islet cell gene expression for each mouse at the time of sacrifice \citep{keller2018genetic}. They shared these data, together with inferred founder allele probabilities, on the Data Dryad site (\url{https://datadryad.org/resource/doi:10.5061/dryad.pj105}). We performed analyses with the R statistical computing environment \citep{r} and the packages \texttt{qtl2} \citep{qtl2} and \texttt{qtl2pleio} \citep{qtl2pleio}.


We study below 80 Chromosome 19 local expression QTL and their corresponding transcript levels. We define a local expression QTL to be an expression QTL that is on the same chromosome as the gene itself. For example, the \emph{Asah2} gene is located on Chromosome 19 and its transcript levels have an expression QTL on Chromosome 19 (Table \ref{tab:ann4}). Thus, we term the Chromosome 19 \emph{Asah2} expression QTL a local expression QTL.

We choose to focus on local expression QTL, while ignoring nonlocal expression QTL, because we know, approximately, the true locations for local expression QTL. That is, a local expression QTL is near the corresponding gene position. Additionally, we expect that a given local expression QTL affects only one local expression trait. In our example above, we expect that the \emph{Asah2} expression QTL is near the \emph{Asah2} gene position and that no other local expression traits map to it.


Our design involves selection of a set of ``anchor'' expression traits. Gene \emph{Asah2} is located near the center of Chromosome 19 and has a very strong local expression QTL (Table \ref{tab:ann4}). We chose it as our first ``anchor'' gene expression trait. To diversify our collection of anchor genes, we chose three additional expression traits with local expression QTL. These three are \emph{Lipo1}, \emph{Lipo2}, and \emph{4933413C19Rik} (Table \ref{tab:ann4}). Together, the four anchor genes represent a variety of strong local expression trait LOD scores (from 60 to 101) and demonstrate modest variability in their founder allele effects (Table~\ref{tab:effects}). All four anchor genes are located near the middle of Chromosome 19 (Table~\ref{tab:ann4}).




We identified a set of 76 non-anchor local expression traits that map to the 20-Mb region centered on the peak for \emph{Asah2}, at 32.1 Mb. Each trait among the 76 maps to Chromosome 19 with a univariate LOD score of at least 10 (Table \ref{tab:ann76}).


% latex table generated in R 3.5.2 by xtable 1.8-3 package
% Thu Jan  3 16:29:15 2019
\begin{table}[ht]
\caption{Annotations for four anchor genes.}\label{tab:ann4}
\centering
\begin{tabular}{>{\em}lrrrr}
  \hline
symbol & start & end & peak\_position & lod \\
  \hline
Asah2 & 31.98 & 32.06 & 32.14 & 101.20 \\
  Lipo1 & 33.52 & 33.76 & 33.67 & 85.46 \\
  Lipo2 & 33.72 & 33.76 & 33.02 & 77.21 \\
  4933413C19Rik & 28.58 & 28.58 & 28.78 & 60.41 \\
   \hline
\end{tabular}
\end{table}







For each Chromosome 19 marker, we estimated founder allele and covariate effects. We calculated:

\begin{equation}\label{eq:uni-Bhat}
\widehat {(B:C)} = \left((X:W)^T\hat\Gamma^{-1}(X:W)\right)^{-1}(X:W)^T\hat\Gamma^{-1}Y
\end{equation}

\noindent where $B:C$ denotes the concatenation of $B$ and $C$, and $X:W$ refers to the $n$ by $12$ matrix resulting from appending the columns of $W$ to the $X$ matrix. $\hat\Gamma$ is the covariance matrix defined by Equation \ref{eq:Sigma}.

\begin{equation}\label{eq:Sigma}
\hat \Gamma = \hat\sigma_g^2 K + \hat \sigma_e^2 I_n
\end{equation}

\noindent We denote the restricted maximum likelihood estimates of the variance components by $\hat \sigma_g^2$ and $\hat \sigma_e^2$.


For each of the 80 expression traits, we calculated fitted values for each subject with the estimated founder allele and covariate effects (Equation \ref{eq:uni-Bhat}). We then calculated correlations between fitted values for pairs of traits. Each pairing involved one anchor gene expression trait and one other gene expression trait.

We anticipated that more similar two traits' founder allele effects would correspond, on average, to smaller pleiotropy test statistics. We base this expectation on findings from \citet{macdonald2007joint} and \citet{king2012genetic}. \citet{macdonald2007joint} and \citet{king2012genetic} found that two traits that associate with a single pleiotropic QTL tended to have similar founder allele effects patterns for biallelic markers.

We performed two-dimensional QTL scans for $4 * 76 + \binom{4}{2} = 310$ pairs. Each pair included one of the four anchor gene expression traits and either one of 76 non-anchor gene expression traits or one of the remaining three anchor gene expression traits.

Our two-dimensional QTL scan encompassed a 1000 by 1000 marker grid from 18.1 Mb to 42.5 Mb on Chromosome 19. Each scan involved fitting 1000 x 1000 = 1,000,000 models via generalized least squares. For a given ordered pair of markers, we used the bivariate linear mixed effects model and methods defined in Chapter 2. These methods are implemented in the R package \texttt{qtl2pleio} \citep{qtl2pleio}.

For each of the 80 expression traits, we used the fitted founder allele and covariate effects ($\widehat{B:C}$ in Equation~\ref{eq:uni-Bhat}) to obtain fitted values vectors for every subject and all 80 traits (Equation~\ref{eq:fitted}). For each of the 310 pairings of traits, we calculated correlations among the fitted values vectors. Our motivation for working with the fitted values vectors (instead of the $\hat B$ estimated founder allele effects vectors) is that the fitted values approximately weight the allele effects by allele frequency. An alternative analysis might neglect the covariates when calculating fitted values.

\begin{equation}\label{eq:fitted}
\hat Y = X\hat B + W \hat C
\end{equation}


\subsection{Results}

% latex table generated in R 3.5.2 by xtable 1.8-3 package
% Sat Feb  2 10:17:32 2019
\begin{table}[ht]
\caption{Founder allele effect estimates at Chromosome 19 QTL peak position.}\label{tab:effects}
\centering
\begin{tabular}{>{\em}cccc}
  \hline
 Gene Symbol & Founder allele & Effect & Standard error \\
  \hline
\multirow{8}{*}{Asah2} & A & -0.96 & 0.17 \\
  & B & 1.01 & 0.19 \\
  & C & 0.14 & 0.17 \\
  & D & -1.16 & 0.17 \\
  & E & 1.05 & 0.16 \\
  & F & -0.61 & 0.20 \\
  & G & 1.81 & 0.16 \\
  & H & -0.18 & 0.18 \\
  \hline
  \multirow{8}{*}{Lipo1} & A & 0.29 & 0.18 \\
  & B & 0.13 & 0.21 \\
  & C & 0.28 & 0.20 \\
  & D & 0.23 & 0.19 \\
  & E & -0.17 & 0.18 \\
  & F & -0.28 & 0.21 \\
  & G & 2.55 & 0.19 \\
  & H & -0.72 & 0.19 \\
  \hline
\multirow{8}{*}{Lipo2} & A & -0.10 & 0.18 \\
  & B & -0.28 & 0.23 \\
  & C & 0.00 & 0.20 \\
  & D & 0.01 & 0.18 \\
  & E & -0.77 & 0.17 \\
  & F & -0.89 & 0.22 \\
  & G & 2.65 & 0.18 \\
  & H & -0.70 & 0.20 \\
   \hline
\multirow{8}{*}{4933413C19Rik} & A & 0.29 & 0.23 \\
  & B & 0.76 & 0.24 \\
  & C & 0.81 & 0.21 \\
  & D & 0.49 & 0.24 \\
  & E & 0.67 & 0.20 \\
  & F & -0.53 & 0.22 \\
  & G & -1.65 & 0.18 \\
  & H & 0.67 & 0.21 \\
   \hline
\end{tabular}
\end{table}


All four anchor traits demonstrate strong PWK (``G'') allele effects. \emph{Lipo2} and \emph{Asah2} have similar patterns among allele effects (at their respective QTL peaks) (Table~\ref{tab:effects}).


%\subsection{Pleiotropy likelihood ratio test statistic vs. chromosomal position}

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{../keller-2018-chr19-power/Rmd/lrt-v-middle-of-gene.pdf}
    \caption[Pleiotropy LRT vs. chromosomal position plots reveal that higher values of pleiotropy LRT tend to correspond to greater interlocus distance and greater univariate LOD score.]{Each anchor gene has its own panel. Along the horizontal axis is Chromosome 19 position. The vertical axis is for pleiotropy test statistic value. Each point corresponds to a local gene expression trait. Point color corresponds to the nonlocal gene's univariate LOD score, with lighter shades of blue denoting greater values of univariate LOD score. Vertical black bar denotes the anchor gene's position on Chromosome 19. All four panels reveal that points further from the anchor gene tend to show greater test statistic values. Additionally, the Lipo1 and Lipo2 panels offer an opportunity to compare the impact of anchor gene univariate LOD score on pleiotropy test statistic values.}
    \label{fig:middle}
\end{figure}

Each anchor gene has its own panel in Figure \ref{fig:middle}. Along the horizontal axis is Chromosome 19 position. The vertical axis is for pleiotropy test statistics. Each point corresponds to a local gene expression trait. Point color corresponds to the nonlocal gene's univariate LOD score, with lighter shades of blue denoting greater values of univariate LOD score. Vertical black bar denotes the anchor gene's position on Chromosome 19. All four panels reveal that points further from the anchor gene tend to show greater test statistic values. Additionally, because of their nearly identical positions, the \emph{Lipo1} and \emph{Lipo2} panels offer an opportunity to compare the impact of anchor gene univariate LOD score on pleiotropy test statistics.




%\subsection{Pleiotropy likelihood ratio test statistics vs. univariate LOD scores}

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{../keller-2018-chr19-power/Rmd/lrt-v-univariate-lod.pdf}
    \caption[Pleiotropy LRT vs. univariate LOD score plots reveal that greater univariate LOD scores (and greater interlocus distance) tend to correspond to greater pleiotropy LRT values.]{Vertical axis denotes pleiotropy test statistic value, while horizontal axis denotes univariate LOD score. Each point corresponds to a single gene expression trait. Panels correspond to the anchor gene expression trait. The pleiotropy test statistics correspond to analyses involving a single gene expression trait and the specified anchor gene expression trait.}
    \label{fig:lod}
\end{figure}

Analyses for all four anchor gene expression traits demonstrate that greater univariate LOD scores tend to correspond to greater values of the pleiotropy test statistic (Figure \ref{fig:lod}).








%\subsection{Pleiotropy likelihood ratio test statistics vs. fitted values correlation}

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{../keller-2018-chr19-power/Rmd/lrt-v-corr.pdf}
    \caption[Pleiotropy LRT vs. fitted values correlations plots reveal little evidence for a relationship.]{Vertical axis denotes the pleiotropy test statistic value, and horizontal axis indicates absolute value of the correlation between vectors of fitted values. Each point corresponds to a pairing between the specified anchor expression trait and one of the 79 other expression traits.}
    \label{fig:cor}
\end{figure}

Figure \ref{fig:cor} features four panels, one for each anchor gene. Each point corresponds to a pairing between the specified anchor and one of the 79 other gene expression traits.



\subsection{Discussion}

Our goal for this study was to characterize the impacts of univariate LOD score, inter-locus distance, and founder allele effects pattern similarities on pleiotropy test statistic values. Our study design, in which we examined 310 pairs of local gene expression traits on Chromosome 19, allowed us to interrogate both the effects of univariate association strength and the effects of inter-locus distance. We found that stronger univariate associations and greater inter-locus distances correspond to greater pleiotropy test statistic values (Figures \ref{fig:middle} and \ref{fig:lod}). We expected these trends based on our simulation studies in Chapter 2.

Figure \ref{fig:cor} revealed no marginal relationship between fitted values correlations and pleiotropy test statistics. However, close examination of Figure \ref{fig:cor} reveals the possibility that there is an interaction between 1) fitted values correlations and 2) univariate association strength. In every panel, those expression traits with stronger univariate associations tend to have steeper slopes between the conditional mean pleiotropy test statistic values and fitted values correlations. The plots suggest that, at greater univariate LOD values, there is a greater (negative) relationship between fitted values correlation and pleiotropy test statistic value.

We anticipated that more similar founder allele effects patterns would correspond to smaller values for the pleiotropy test statistic, when holding other factors constant. As we stated above, \citet{macdonald2007joint} and \citet{king2012genetic} argued that, for biallelic markers, two pleiotropic traits should have similar founder allele effects patterns. In our setting, it's unclear whether the markers are biallelic in the collection of eight founder lines.

We've demonstrated strong evidence in support of the roles of 1) univariate QTL LOD scores and 2) interlocus distances impacting pleiotropy test statistic values. Greater univariate QTL scores and greater interlocus distance lead to greater pleiotropy test statistics. Future research may clarify the impoact of founder allele effects patterns on pleiotropy test statistics. The fact that all four anchor traits had strong PWK effects limited our ability to fully define the impact of allele effects patterns on our test statistics.

Throughout this study, we elected to use test statistic values rather than p-values, as our measure of evidence supporting the separate QTL hypothesis. The primary reason for doing this is to avoid the computationally costly bootstrap sampling and two-dimensional QTL scans that we would need to get bootstrap p-values.

We share our analysis \texttt{R} code \citep{r} as a \texttt{git} repository at this URL: \url{https://github.com/fboehm/keller-2018-chr19-power}.

% latex table generated in R 3.5.2 by xtable 1.8-3 package
% Thu Jan  3 16:43:33 2019
\begin{table}[ht]
\caption{Annotations for 76 non-anchor genes on Chromosome 19.}\label{tab:ann76}
\centering
\begingroup\tiny
\begin{tabular}{>{\em}lrrrr}
  \hline
Gene & Start & End & Peak position & LOD \\
  \hline
C030046E11Rik & 29.52 & 29.61 & 29.55 & 95.58 \\
  Tctn3 & 40.60 & 40.61 & 40.59 & 90.00 \\
  Gm7237 & 33.41 & 33.42 & 33.67 & 74.61 \\
  Lipo4 & 33.50 & 33.52 & 34.00 & 68.23 \\
  Dock8 & 25.00 & 25.20 & 25.07 & 63.17 \\
  Sorbs1 & 40.30 & 40.40 & 40.48 & 61.89 \\
  Lipm & 34.10 & 34.12 & 34.06 & 58.43 \\
  Blnk & 40.93 & 40.99 & 40.76 & 57.16 \\
  A830019P07Rik & 35.84 & 35.92 & 35.60 & 55.54 \\
  Uhrf2 & 30.03 & 30.09 & 29.96 & 54.40 \\
  Mbl2 & 30.23 & 30.24 & 30.18 & 52.81 \\
  Myof & 37.90 & 38.04 & 38.05 & 48.46 \\
  Gm27042 & 40.59 & 40.59 & 40.61 & 44.27 \\
  Btaf1 & 36.93 & 37.01 & 36.90 & 41.25 \\
  Hoga1 & 42.05 & 42.07 & 42.09 & 41.23 \\
  Ppp1r3c & 36.73 & 36.74 & 36.53 & 40.69 \\
  Pcgf5 & 36.38 & 36.46 & 36.24 & 40.06 \\
  Slc35g1 & 38.40 & 38.41 & 38.35 & 38.11 \\
  Pten & 32.76 & 32.83 & 32.77 & 37.95 \\
  Gldc & 30.10 & 30.18 & 30.17 & 36.26 \\
  Lgi1 & 38.26 & 38.31 & 38.17 & 34.91 \\
  C330002G04Rik & 23.04 & 23.08 & 23.34 & 34.84 \\
  Ppapdc2 & 28.96 & 28.97 & 29.09 & 34.71 \\
  Gm8978 & 33.61 & 33.63 & 33.03 & 34.59 \\
  Mms19 & 41.94 & 41.98 & 41.98 & 32.03 \\
  Ankrd22 & 34.12 & 34.17 & 34.04 & 31.83 \\
  Cdc37l1 & 28.99 & 29.02 & 29.03 & 31.14 \\
  Sgms1 & 32.12 & 32.39 & 32.11 & 30.10 \\
  Entpd1 & 40.61 & 40.74 & 40.50 & 29.73 \\
  Cbwd1 & 24.92 & 24.96 & 24.73 & 29.65 \\
  Gm14446 & 34.59 & 34.60 & 34.28 & 27.65 \\
  Ermp1 & 29.61 & 29.65 & 29.70 & 26.57 \\
  Gm9938 & 23.72 & 23.73 & 23.87 & 26.46 \\
  Insl6 & 29.32 & 29.33 & 29.37 & 26.23 \\
  Slc16a12 & 34.67 & 34.75 & 34.71 & 25.54 \\
  Pgm5 & 24.68 & 24.86 & 25.00 & 24.30 \\
  Morn4 & 42.07 & 42.09 & 41.79 & 23.86 \\
  Exosc1 & 41.92 & 41.93 & 42.10 & 23.28 \\
  Smarca2 & 26.61 & 26.78 & 26.59 & 23.25 \\
  4930418C01Rik & 24.42 & 24.43 & 23.92 & 23.10 \\
  2700046G09Rik & 32.39 & 32.39 & 32.25 & 23.02 \\
  Kcnv2 & 27.32 & 27.34 & 27.14 & 22.88 \\
  1500017E21Rik & 36.61 & 36.71 & 37.07 & 22.78 \\
  Fra10ac1 & 38.19 & 38.22 & 38.35 & 22.48 \\
  Rnls & 33.14 & 33.39 & 34.17 & 21.94 \\
  Noc3l & 38.79 & 38.82 & 40.20 & 21.67 \\
  Pip5k1b & 24.29 & 24.56 & 24.15 & 21.62 \\
  Plgrkt & 29.35 & 29.37 & 29.37 & 20.65 \\
  Ifit3 & 34.58 & 34.59 & 34.28 & 20.45 \\
  Fas & 34.29 & 34.33 & 34.20 & 19.65 \\
  Slit1 & 41.60 & 41.74 & 41.70 & 18.95 \\
  Rrp12 & 41.86 & 41.90 & 41.71 & 18.09 \\
  Ak3 & 29.02 & 29.05 & 29.55 & 16.90 \\
  A1cf & 31.87 & 31.95 & 32.11 & 15.56 \\
  4430402I18Rik & 28.90 & 28.97 & 29.37 & 15.43 \\
  Pdlim1 & 40.22 & 40.27 & 40.25 & 15.25 \\
  Gm26902 & 34.47 & 34.48 & 36.15 & 14.26 \\
  Plce1 & 38.48 & 38.79 & 38.42 & 14.26 \\
  Slc1a1 & 28.84 & 28.91 & 28.97 & 14.18 \\
  Fam122a & 24.48 & 24.48 & 24.08 & 14.07 \\
  Lipa & 34.49 & 34.53 & 34.29 & 14.06 \\
  Mamdc2 & 23.30 & 23.45 & 23.35 & 13.12 \\
  Kif11 & 37.38 & 37.42 & 37.33 & 12.93 \\
  4933411K16Rik & 42.05 & 42.05 & 42.08 & 12.92 \\
  Ccnj & 40.83 & 40.85 & 40.59 & 12.19 \\
  Gm340 & 41.58 & 41.59 & 41.30 & 12.17 \\
  Fxn & 24.26 & 24.28 & 24.31 & 12.07 \\
  Stambpl1 & 34.19 & 34.24 & 34.28 & 11.62 \\
  Pde6c & 38.13 & 38.18 & 38.07 & 11.54 \\
  Cyp26a1 & 37.70 & 37.70 & 37.48 & 11.35 \\
  Ch25h & 34.47 & 34.48 & 32.50 & 10.74 \\
  Pank1 & 34.81 & 34.88 & 35.55 & 10.61 \\
  9930021J03Rik & 29.71 & 29.81 & 28.71 & 10.32 \\
  Klf9 & 23.14 & 23.17 & 23.34 & 10.26 \\
  Ubtd1 & 41.98 & 42.03 & 41.71 & 10.25 \\
  Lipk & 34.01 & 34.05 & 34.29 & 10.23 \\
   \hline
\end{tabular}
\endgroup
\end{table}


%%%% 3B end

\printbibliography

\end{document}